src/proto/h2/mod.rs

```diff
@@ -13,13 +13,33 @@ pub(crate) use self::client::Client;
 pub(crate) use self::server::Server;
 
 fn strip_connection_headers(headers: &mut HeaderMap) {
-    if headers.remove(TRANSFER_ENCODING).is_some() {
-        trace!("removed illegal Transfer-Encoding header");
-    }
-    if headers.contains_key(CONNECTION) {
-        warn!("Connection header illegal in HTTP/2");
-        //TODO: actually remove it, after checking the value
-        //and removing all related headers
+    let connection_headers = [
+        CONNECTION,
+        TRANSFER_ENCODING,
+        UPGRADE,
+        HeaderName::from_lowercase(b"keep-alive").unwrap(),
+        HeaderName::from_lowercase(b"proxy-connection").unwrap(),
+    ];
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection has a more complete set of headers.

Reply:
Updated this list to remove `CONNECTION`, and include the other headers mentioned in the mozilla docs you linked.